### PR TITLE
ref: Unused entity and implicit cast compiler warnings

### PR DESCRIPTION
**Pull request by JanNash**
## :scroll: Description

This PR
- removes the unused `static const char g_hexNybbles[]` in SentryCrashReport.c
- sets the build setting GCC_WARN_64_TO_32_BIT_CONVERSION to YES on project level, removes its overrides on Framework target level
- makes two implicit casts of NSInteger to int explicit (SentryCrashDefaultBinaryImageProvider.m, SentryStacktraceBuilder.m)

## :bulb: Motivation and Context

These three compiler warnings kept showing up in an iOS project I recently started and they kept annoying me (I like my buildtime issues list clean 😏 ) so I thought I'd just go and fix them. After I forked, cloned, set up and built the framework, only the unused entity warning appeared on build, not so the implicit cast warning. I checked which build setting is required to activate the respective checks and activated it.

## :green_heart: How did you test it?

By: 
- building the framework after adding the build setting to make sure the warnings did now appear
- building the framework after making the casts explicit make sure the warnings are gone
- running the tests

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify the changes
- [ ] I've updated the CHANGELOG
- [x] No breaking changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## unreleased
 
+- fix: Set GCC_WARN_64_TO_32_BIT_CONVERSION to YES on project level
+- fix: Fix compiler warnings regarding unused entity and implicit casts
```
Comment (philipphofmann) [inline CHANGELOG.md:6]: `h`: As this is a refactoring for us we don't mention it in the ReadMe. So let's please remove it.
```suggestion
```

**Comment by JanNash** [comment]
You're welcome :). 
Sorry for not complying with the commit guidelines. Could you point out the specific problem with my commits regarding the guidelines?

**Comment by philipphofmann** [comment]
> You're welcome :).
> Sorry for not complying with the commit guidelines (I have read them but seemingly not thoroughly enough). Could you point out the specific problem with my commits regarding the guidelines?

No problem I can help you to get it right. We squash our PRs so we have a clean git history on our master branch. So in an optimal world, you open up a PR with just a single commit that complies with our [commit message format guidelines](https://develop.sentry.dev/code-review/#commit-message-format). Then Github takes the first line of your commit message and turns this into the title of the PR and the other lines go to the description. When we merge the PR we squash it and make sure the squash commit is compliant with our guidelines. So if you add a few commits afterward they don't have to follow the commit guidelines, because we anyway can modify the squash commit message in Github.

For this PR we need to prefix the title with the type of the change. Therefore I prefixed it with `ref` which stands for refactoring.

TLDR;
Don't worry too much about it if you don't plan to commit to Sentry too often. We can help you to get it right.


One more thing, your branch needs to be up to date with our base branch before we can merge it. So please merge the latest master into your branch. Then we can merge it.

**Comment by JanNash** [comment]
Thank you for the quick and elaborate response, that all makes sense. I've merged master into this branch, so I guess we'll just have wait for Travis one more time. Have a nice day 👍

**Comment by philipphofmann** [comment]
> Thank you for your quick and elaborate response, that all makes sense. I've merged master into this branch, so I guess we'll just have wait for Travis one more time. Have a nice day 👍

You are welcome. Thanks again for the help. You to have a nice day 👏
